node/src/components/chainspec_loader.rs

```diff
@@ -373,8 +373,25 @@ impl ChainspecLoader {
             if highest_block.header().is_switch_block() {
                 // This is a valid run immediately after upgrading the node version.
                 trace!("valid run immediately after upgrade");
+
+                // TODO - remove this once we have protocol versions in block headers.
+                //
+                // It is required if we restart after the network has had a hard reset upgrade from
+                // delta currently running at v1.0.0.  In that case, the cached version could well
+                // be the same as that in the current chainspec.  However, we need to pass the
+                // previous version when calling `ContractRuntime::upgrade`.
+                //
+                // If we end up running the release-0.8.0, we'll need to update this workaround to
+                // allow for upgrading from 0.8.0.
```

Comment:
Have you created the JIRA ticket to remove it?

Reply:
Well, no.  I don't think this will get removed.  What I meant was that if we end up running delta using `release-0.8.0` and set up a new release branch for _its_ next version, then we'll need to extend this `if` block to cater for that version upgrade too.

This won't need removed from master/dev since it should never end up ported there.